docs/design/mp_arktos.md

```diff
@@ -0,0 +1,241 @@
+* Update object store: map subnet object (which is equivalent to net object in Mizar) with pod: ```self.store.eps_net_store[net1][nginx] = pod object```
+* After Pod creation is completed, update Pod object with new annotations: ```mizar.futurewei.com/network_user_input``` and ```arktos.futurewei.com/network-readiness```.      
+* ```mizar.futurewei.com/network_user_input``` will include Mizar network information such as ```vpc name```, ```subnet name``` and ```ip address```
+* In cni daemon service, it will retrieve newly created Pod object, and pass in required network configuration. See below: 
+
+```
+    def cni_add(self, params):
+        ep = self.get_pod_obj(params)
+
+        result = {
+            "cniVersion": params.cni_version,
+            "interfaces": [
+                {
+                    "name": pod.veth_name,
+                    "mac": pod.mac,
+                    "sandbox": pod.netns
+                }
+            ],
+            "ips": [
+                {
+                    "version": "4",
+                    "address": "{}/{}".format(pod.ip, pod.prefix),
+                    "gateway": pod.gw,
+                    "interface": 0
+                }
+            ]
+        }
+
+    def get_pod_obj(self, params):
+        logger.debug("Allocate endpoint name")
+        name = ""
+        if 'K8S_POD_NAME' in params.cni_args_dict:
+            name = params.cni_args_dict['K8S_POD_NAME']
+        
+        if CniService.store.contains_ep(name):
+            return CniService.store.get_ep(name)
+```
+
+* Once creation is completed, operator will set ```arktos.futurewei.com/network-readiness == true```. 
+
+* In return, the annotations becomes: 
+```yaml
+  annotations:
+    arktos.futurewei.com/nic: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+    arktos.futurewei.com/vip-hint: {"cidr":"10.10.1.0/26", "ip":"10.10.1.12", "policy":"BestEffort"}
+    arktos.futurewei.com/network-readiness: "true"
+    mizar.futurewei.com/network_user_input: {"vpc": "vpc-1", "net": "net1", "ip": "10.10.1.12"}
+```
+Note: Integrity checks are needed here, where we need make sure that the ip address falls within subnet CIDR range and service CIDR range, and subnet CIDR range falls within the VPC's (vpc-1) CIDR range.
+
+
+### Service Creation 
+
+* List Kubernetes cluster object / Arktos newtork object
+* Waiting on customer to create service using yaml file:  
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  labels:
+      arktos.futurewei.com/network: vpc-1
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+* Note that if ```arktos.futurewei.com/network``` not listed, and default network is selected
+* Build-in operator kicks off ```my-service``` creation
+* Use endpoint operator to create a scaled endpoint object within ```vpc-1```. 
```

Comment:
if the service ip is decided by mizar, please update spec.ClusterIP it in arktos service object.

Reply:
No problem, this is something we already have in our current design.